assets/src/stories-editor/components/higher-order/with-call-to-action-validation.js

```diff
@@ -1,61 +1,64 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
 /**
  * WordPress dependencies
  */
 import { getBlockType } from '@wordpress/blocks';
 import { Button } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { Warning } from '@wordpress/block-editor';
-import { createHigherOrderComponent, compose } from '@wordpress/compose';
+import { createHigherOrderComponent } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { isCTABlock } from '../../helpers';
 
-const enhance = compose(
-	/**
-	 * For blocks which are only allowed once per page,  provides the
-	 * wrapped component with `originalBlockClientId` -- a reference to the
-	 * first block of the same type on the page -- if and only if that
-	 * "original" block is not the current one. Thus, an non-existent
-	 * `originalBlockClientId` prop signals that the block is valid.
-	 */
-	withSelect( ( select, props ) => {
-		const { getBlockRootClientId, getBlock, getBlockOrder, getBlocksByClientId } = select( 'core/block-editor' );
-
-		if ( ! isCTABlock( props.name ) ) {
-			return {};
-		}
+export default createHigherOrderComponent( ( BlockEdit ) => {
+	const CallToActionValidation = ( props ) => {
+		const {
+			isInvalid,
+			originalBlockClientId,
```

Comment:
The value of this isn't used at all - only its presence. Can this change to a properly named boolean in stead, maybe `hasDuplicate` or similar?

Reply:
It is used for the `selectBlock( originalBlockClientId )` part (when clicking on "Find original")

Reply:
Ah yes, my bad.

But twice it's used for its _truthiness_ only, maybe a `const hasDuplicate = Boolean( originalBlockClientId );` would make things more obvious for those.